{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "5144ca85-7444-4bfa-a230-8d95b5ca0318",
   "metadata": {},
   "source": [
    "Gene-level Analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "d8e519c1-890f-49cc-8b6f-f104ff2ea359",
   "metadata": {},
   "outputs": [],
   "source": [
    "# imports\n",
    "import pandas as pd\n",
    "import os\n",
    "from pprint import pprint"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "id": "b1b06b0c-b43b-45a7-866b-e6448570dfb4",
   "metadata": {},
   "outputs": [],
   "source": [
    "# connect to file server\n",
    "def make_query(  url       = \"http://45.79.5.6:3000/clean\",\n",
    "                 consort   = \"EBOPLUS\",\n",
    "                 cohort    = \"usa\",\n",
    "                 dtlevel   = \"gene_expression\",\n",
    "                 dtset     = \"rnaseq\",\n",
    "                 dttype    = \"countsNormCPM\" ):\n",
    "    fdir  = os.path.join(url, consort, cohort, dtlevel, dtset)\n",
    "    fname = dttype+\"_\"+cohort+\".csv\"\n",
    "    query = os.path.join(fdir, fname)\n",
    "    return query"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "472e10cc-edba-47c8-9eb0-6babddae4bf0",
   "metadata": {},
   "source": [
    "## Data wrangling for plots"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b52def72-92de-46d3-a2e5-6d8efe1a1f80",
   "metadata": {},
   "source": [
    "### Gene Expression Barplot"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f9168023-eef4-4d17-a7b0-314b6a367776",
   "metadata": {},
   "source": [
    "**INPUT**: \n",
    "\n",
    "*Exprs data*: | sample_id | GeneID | gene expression |   \n",
    "\n",
    "*Pheno*:      | sample_id | pheno-var |"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "620ea937-2d99-4e47-9346-78a0c8f482f9",
   "metadata": {},
   "source": [
    "**OUTPUT** \n",
    "\n",
    "*format: json*\n",
    "\n",
    "- GeneID\n",
    "- gene expression\n",
    "- sample_id\n",
    "- group-var"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "8f859c58-fcf1-468f-9bcf-9e3a2a892092",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>S001_045_D1</th>\n",
       "      <th>S001_046_D0</th>\n",
       "      <th>S001_050_D3</th>\n",
       "      <th>S001_053_D0</th>\n",
       "      <th>S001_059_D1</th>\n",
       "      <th>S001_059_D2</th>\n",
       "      <th>S001_060_D0</th>\n",
       "      <th>S001_043_D0</th>\n",
       "      <th>S001_043_D1</th>\n",
       "      <th>...</th>\n",
       "      <th>S002_058_D1</th>\n",
       "      <th>S004_052_D0</th>\n",
       "      <th>S008_051_D3</th>\n",
       "      <th>S008_059_D0</th>\n",
       "      <th>S008_059_D1</th>\n",
       "      <th>S008_059_D7</th>\n",
       "      <th>S008_059_D3</th>\n",
       "      <th>S002_050_D7</th>\n",
       "      <th>S002_050_D1</th>\n",
       "      <th>S002_058_D3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>-1.533514</td>\n",
       "      <td>-2.365695</td>\n",
       "      <td>-1.448913</td>\n",
       "      <td>-2.187106</td>\n",
       "      <td>-1.211852</td>\n",
       "      <td>-3.141005</td>\n",
       "      <td>-2.101967</td>\n",
       "      <td>-1.336650</td>\n",
       "      <td>-3.141005</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.191881</td>\n",
       "      <td>-0.928434</td>\n",
       "      <td>0.146730</td>\n",
       "      <td>-0.547439</td>\n",
       "      <td>-3.141005</td>\n",
       "      <td>-0.172061</td>\n",
       "      <td>0.616546</td>\n",
       "      <td>-1.119818</td>\n",
       "      <td>-1.769805</td>\n",
       "      <td>-0.840153</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A2M</td>\n",
       "      <td>0.590535</td>\n",
       "      <td>0.718924</td>\n",
       "      <td>-0.691444</td>\n",
       "      <td>-2.187106</td>\n",
       "      <td>-1.875473</td>\n",
       "      <td>-2.244887</td>\n",
       "      <td>-2.101967</td>\n",
       "      <td>1.687831</td>\n",
       "      <td>0.691224</td>\n",
       "      <td>...</td>\n",
       "      <td>0.711553</td>\n",
       "      <td>1.250799</td>\n",
       "      <td>2.447441</td>\n",
       "      <td>1.823177</td>\n",
       "      <td>0.876028</td>\n",
       "      <td>1.682669</td>\n",
       "      <td>2.031717</td>\n",
       "      <td>2.171863</td>\n",
       "      <td>-0.614794</td>\n",
       "      <td>1.877802</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A4GALT</td>\n",
       "      <td>-3.141005</td>\n",
       "      <td>-1.864142</td>\n",
       "      <td>-3.141005</td>\n",
       "      <td>-3.141005</td>\n",
       "      <td>-3.141005</td>\n",
       "      <td>-2.244887</td>\n",
       "      <td>-3.141005</td>\n",
       "      <td>-3.141005</td>\n",
       "      <td>-3.141005</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.809392</td>\n",
       "      <td>-1.646588</td>\n",
       "      <td>-3.141005</td>\n",
       "      <td>-0.940293</td>\n",
       "      <td>-0.724164</td>\n",
       "      <td>-0.998652</td>\n",
       "      <td>-0.553645</td>\n",
       "      <td>0.792205</td>\n",
       "      <td>2.147521</td>\n",
       "      <td>-1.160755</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AAAS</td>\n",
       "      <td>2.699418</td>\n",
       "      <td>3.087852</td>\n",
       "      <td>2.705112</td>\n",
       "      <td>2.467571</td>\n",
       "      <td>2.623416</td>\n",
       "      <td>2.985887</td>\n",
       "      <td>2.635039</td>\n",
       "      <td>2.880921</td>\n",
       "      <td>2.613185</td>\n",
       "      <td>...</td>\n",
       "      <td>3.210792</td>\n",
       "      <td>3.062590</td>\n",
       "      <td>3.364397</td>\n",
       "      <td>3.770955</td>\n",
       "      <td>2.952177</td>\n",
       "      <td>3.503279</td>\n",
       "      <td>3.045054</td>\n",
       "      <td>3.329799</td>\n",
       "      <td>3.158312</td>\n",
       "      <td>3.207962</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AACS</td>\n",
       "      <td>3.384141</td>\n",
       "      <td>2.211161</td>\n",
       "      <td>2.114024</td>\n",
       "      <td>2.548390</td>\n",
       "      <td>2.765276</td>\n",
       "      <td>2.712725</td>\n",
       "      <td>3.022402</td>\n",
       "      <td>2.683533</td>\n",
       "      <td>2.780137</td>\n",
       "      <td>...</td>\n",
       "      <td>3.060751</td>\n",
       "      <td>2.745547</td>\n",
       "      <td>2.157297</td>\n",
       "      <td>3.047658</td>\n",
       "      <td>3.483244</td>\n",
       "      <td>2.991366</td>\n",
       "      <td>2.792562</td>\n",
       "      <td>2.559054</td>\n",
       "      <td>2.508396</td>\n",
       "      <td>2.787227</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 158 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  gene_id  S001_045_D1  S001_046_D0  S001_050_D3  S001_053_D0  S001_059_D1  \\\n",
       "0    A1BG    -1.533514    -2.365695    -1.448913    -2.187106    -1.211852   \n",
       "1     A2M     0.590535     0.718924    -0.691444    -2.187106    -1.875473   \n",
       "2  A4GALT    -3.141005    -1.864142    -3.141005    -3.141005    -3.141005   \n",
       "3    AAAS     2.699418     3.087852     2.705112     2.467571     2.623416   \n",
       "4    AACS     3.384141     2.211161     2.114024     2.548390     2.765276   \n",
       "\n",
       "   S001_059_D2  S001_060_D0  S001_043_D0  S001_043_D1  ...  S002_058_D1  \\\n",
       "0    -3.141005    -2.101967    -1.336650    -3.141005  ...    -0.191881   \n",
       "1    -2.244887    -2.101967     1.687831     0.691224  ...     0.711553   \n",
       "2    -2.244887    -3.141005    -3.141005    -3.141005  ...    -0.809392   \n",
       "3     2.985887     2.635039     2.880921     2.613185  ...     3.210792   \n",
       "4     2.712725     3.022402     2.683533     2.780137  ...     3.060751   \n",
       "\n",
       "   S004_052_D0  S008_051_D3  S008_059_D0  S008_059_D1  S008_059_D7  \\\n",
       "0    -0.928434     0.146730    -0.547439    -3.141005    -0.172061   \n",
       "1     1.250799     2.447441     1.823177     0.876028     1.682669   \n",
       "2    -1.646588    -3.141005    -0.940293    -0.724164    -0.998652   \n",
       "3     3.062590     3.364397     3.770955     2.952177     3.503279   \n",
       "4     2.745547     2.157297     3.047658     3.483244     2.991366   \n",
       "\n",
       "   S008_059_D3  S002_050_D7  S002_050_D1  S002_058_D3  \n",
       "0     0.616546    -1.119818    -1.769805    -0.840153  \n",
       "1     2.031717     2.171863    -0.614794     1.877802  \n",
       "2    -0.553645     0.792205     2.147521    -1.160755  \n",
       "3     3.045054     3.329799     3.158312     3.207962  \n",
       "4     2.792562     2.559054     2.508396     2.787227  \n",
       "\n",
       "[5 rows x 158 columns]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "query = make_query( consort=\"EBOPLUS\", cohort=\"usa\", dtlevel=\"gene_expression\", dtset=\"rnaseq\", dttype=\"countsNormCPM\" )\n",
    "data = pd.read_csv(query)\n",
    "data.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "1f5117b8-686f-4c39-afba-75aba24f832f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>dose</th>\n",
       "      <th>sampling_day</th>\n",
       "      <th>mapped_reads</th>\n",
       "      <th>on_target</th>\n",
       "      <th>arthralgia</th>\n",
       "      <th>pain_in_extremity</th>\n",
       "      <th>pyrexia</th>\n",
       "      <th>myalgia</th>\n",
       "      <th>chills</th>\n",
       "      <th>age</th>\n",
       "      <th>gender</th>\n",
       "      <th>volunteer_id</th>\n",
       "      <th>treatment</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>S001_045_D1</td>\n",
       "      <td>20000000.0</td>\n",
       "      <td>1</td>\n",
       "      <td>11192441</td>\n",
       "      <td>88.28</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>48</td>\n",
       "      <td>0</td>\n",
       "      <td>001-045</td>\n",
       "      <td>V</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>S001_046_D0</td>\n",
       "      <td>20000000.0</td>\n",
       "      <td>0</td>\n",
       "      <td>15527796</td>\n",
       "      <td>85.08</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>39</td>\n",
       "      <td>1</td>\n",
       "      <td>001-046</td>\n",
       "      <td>V</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>S001_050_D3</td>\n",
       "      <td>20000000.0</td>\n",
       "      <td>3</td>\n",
       "      <td>10021764</td>\n",
       "      <td>88.52</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>45</td>\n",
       "      <td>0</td>\n",
       "      <td>001-050</td>\n",
       "      <td>V</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>S001_053_D0</td>\n",
       "      <td>20000000.0</td>\n",
       "      <td>0</td>\n",
       "      <td>9626092</td>\n",
       "      <td>85.78</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>1</td>\n",
       "      <td>001-053</td>\n",
       "      <td>V</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>S001_059_D1</td>\n",
       "      <td>20000000.0</td>\n",
       "      <td>1</td>\n",
       "      <td>10523308</td>\n",
       "      <td>88.57</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>001-059</td>\n",
       "      <td>V</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     sample_id        dose  sampling_day  mapped_reads  on_target  arthralgia  \\\n",
       "0  S001_045_D1  20000000.0             1      11192441      88.28           0   \n",
       "1  S001_046_D0  20000000.0             0      15527796      85.08           0   \n",
       "2  S001_050_D3  20000000.0             3      10021764      88.52           0   \n",
       "3  S001_053_D0  20000000.0             0       9626092      85.78           0   \n",
       "4  S001_059_D1  20000000.0             1      10523308      88.57           0   \n",
       "\n",
       "   pain_in_extremity  pyrexia  myalgia  chills  age  gender volunteer_id  \\\n",
       "0                  1        0        0       0   48       0      001-045   \n",
       "1                  1        0        0       0   39       1      001-046   \n",
       "2                  1        1        0       0   45       0      001-050   \n",
       "3                  1        1        1       0   20       1      001-053   \n",
       "4                  0        0        0       0   19       0      001-059   \n",
       "\n",
       "  treatment  \n",
       "0         V  \n",
       "1         V  \n",
       "2         V  \n",
       "3         V  \n",
       "4         V  "
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "query = make_query( consort=\"EBOPLUS\", cohort=\"usa\", dtlevel=\"gene_expression\", dtset=\"rnaseq\", dttype=\"pheno\" )\n",
    "pheno = pd.read_csv(query)\n",
    "pheno.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "459c9f91-9beb-4f07-9a5d-3fd9e8e937b6",
   "metadata": {},
   "outputs": [],
   "source": [
    "# exprs\n",
    "gene_id = \"A1BG\"\n",
    "\n",
    "# pheno\n",
    "pheno_var = \"gender\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "aadbc103-fa6f-4dd1-9dff-4f5bf11c65b5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>S001_045_D1</th>\n",
       "      <th>S001_046_D0</th>\n",
       "      <th>S001_050_D3</th>\n",
       "      <th>S001_053_D0</th>\n",
       "      <th>S001_059_D1</th>\n",
       "      <th>S001_059_D2</th>\n",
       "      <th>S001_060_D0</th>\n",
       "      <th>S001_043_D0</th>\n",
       "      <th>S001_043_D1</th>\n",
       "      <th>S001_043_D2</th>\n",
       "      <th>...</th>\n",
       "      <th>S002_058_D1</th>\n",
       "      <th>S004_052_D0</th>\n",
       "      <th>S008_051_D3</th>\n",
       "      <th>S008_059_D0</th>\n",
       "      <th>S008_059_D1</th>\n",
       "      <th>S008_059_D7</th>\n",
       "      <th>S008_059_D3</th>\n",
       "      <th>S002_050_D7</th>\n",
       "      <th>S002_050_D1</th>\n",
       "      <th>S002_058_D3</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>-1.533514</td>\n",
       "      <td>-2.365695</td>\n",
       "      <td>-1.448913</td>\n",
       "      <td>-2.187106</td>\n",
       "      <td>-1.211852</td>\n",
       "      <td>-3.141005</td>\n",
       "      <td>-2.101967</td>\n",
       "      <td>-1.336650</td>\n",
       "      <td>-3.141005</td>\n",
       "      <td>-3.141005</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.191881</td>\n",
       "      <td>-0.928434</td>\n",
       "      <td>0.146730</td>\n",
       "      <td>-0.547439</td>\n",
       "      <td>-3.141005</td>\n",
       "      <td>-0.172061</td>\n",
       "      <td>0.616546</td>\n",
       "      <td>-1.119818</td>\n",
       "      <td>-1.769805</td>\n",
       "      <td>-0.840153</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M</th>\n",
       "      <td>0.590535</td>\n",
       "      <td>0.718924</td>\n",
       "      <td>-0.691444</td>\n",
       "      <td>-2.187106</td>\n",
       "      <td>-1.875473</td>\n",
       "      <td>-2.244887</td>\n",
       "      <td>-2.101967</td>\n",
       "      <td>1.687831</td>\n",
       "      <td>0.691224</td>\n",
       "      <td>-0.439680</td>\n",
       "      <td>...</td>\n",
       "      <td>0.711553</td>\n",
       "      <td>1.250799</td>\n",
       "      <td>2.447441</td>\n",
       "      <td>1.823177</td>\n",
       "      <td>0.876028</td>\n",
       "      <td>1.682669</td>\n",
       "      <td>2.031717</td>\n",
       "      <td>2.171863</td>\n",
       "      <td>-0.614794</td>\n",
       "      <td>1.877802</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A4GALT</th>\n",
       "      <td>-3.141005</td>\n",
       "      <td>-1.864142</td>\n",
       "      <td>-3.141005</td>\n",
       "      <td>-3.141005</td>\n",
       "      <td>-3.141005</td>\n",
       "      <td>-2.244887</td>\n",
       "      <td>-3.141005</td>\n",
       "      <td>-3.141005</td>\n",
       "      <td>-3.141005</td>\n",
       "      <td>-3.141005</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.809392</td>\n",
       "      <td>-1.646588</td>\n",
       "      <td>-3.141005</td>\n",
       "      <td>-0.940293</td>\n",
       "      <td>-0.724164</td>\n",
       "      <td>-0.998652</td>\n",
       "      <td>-0.553645</td>\n",
       "      <td>0.792205</td>\n",
       "      <td>2.147521</td>\n",
       "      <td>-1.160755</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAAS</th>\n",
       "      <td>2.699418</td>\n",
       "      <td>3.087852</td>\n",
       "      <td>2.705112</td>\n",
       "      <td>2.467571</td>\n",
       "      <td>2.623416</td>\n",
       "      <td>2.985887</td>\n",
       "      <td>2.635039</td>\n",
       "      <td>2.880921</td>\n",
       "      <td>2.613185</td>\n",
       "      <td>1.199839</td>\n",
       "      <td>...</td>\n",
       "      <td>3.210792</td>\n",
       "      <td>3.062590</td>\n",
       "      <td>3.364397</td>\n",
       "      <td>3.770955</td>\n",
       "      <td>2.952177</td>\n",
       "      <td>3.503279</td>\n",
       "      <td>3.045054</td>\n",
       "      <td>3.329799</td>\n",
       "      <td>3.158312</td>\n",
       "      <td>3.207962</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AACS</th>\n",
       "      <td>3.384141</td>\n",
       "      <td>2.211161</td>\n",
       "      <td>2.114024</td>\n",
       "      <td>2.548390</td>\n",
       "      <td>2.765276</td>\n",
       "      <td>2.712725</td>\n",
       "      <td>3.022402</td>\n",
       "      <td>2.683533</td>\n",
       "      <td>2.780137</td>\n",
       "      <td>2.556988</td>\n",
       "      <td>...</td>\n",
       "      <td>3.060751</td>\n",
       "      <td>2.745547</td>\n",
       "      <td>2.157297</td>\n",
       "      <td>3.047658</td>\n",
       "      <td>3.483244</td>\n",
       "      <td>2.991366</td>\n",
       "      <td>2.792562</td>\n",
       "      <td>2.559054</td>\n",
       "      <td>2.508396</td>\n",
       "      <td>2.787227</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 157 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         S001_045_D1  S001_046_D0  S001_050_D3  S001_053_D0  S001_059_D1  \\\n",
       "gene_id                                                                    \n",
       "A1BG       -1.533514    -2.365695    -1.448913    -2.187106    -1.211852   \n",
       "A2M         0.590535     0.718924    -0.691444    -2.187106    -1.875473   \n",
       "A4GALT     -3.141005    -1.864142    -3.141005    -3.141005    -3.141005   \n",
       "AAAS        2.699418     3.087852     2.705112     2.467571     2.623416   \n",
       "AACS        3.384141     2.211161     2.114024     2.548390     2.765276   \n",
       "\n",
       "         S001_059_D2  S001_060_D0  S001_043_D0  S001_043_D1  S001_043_D2  ...  \\\n",
       "gene_id                                                                   ...   \n",
       "A1BG       -3.141005    -2.101967    -1.336650    -3.141005    -3.141005  ...   \n",
       "A2M        -2.244887    -2.101967     1.687831     0.691224    -0.439680  ...   \n",
       "A4GALT     -2.244887    -3.141005    -3.141005    -3.141005    -3.141005  ...   \n",
       "AAAS        2.985887     2.635039     2.880921     2.613185     1.199839  ...   \n",
       "AACS        2.712725     3.022402     2.683533     2.780137     2.556988  ...   \n",
       "\n",
       "         S002_058_D1  S004_052_D0  S008_051_D3  S008_059_D0  S008_059_D1  \\\n",
       "gene_id                                                                    \n",
       "A1BG       -0.191881    -0.928434     0.146730    -0.547439    -3.141005   \n",
       "A2M         0.711553     1.250799     2.447441     1.823177     0.876028   \n",
       "A4GALT     -0.809392    -1.646588    -3.141005    -0.940293    -0.724164   \n",
       "AAAS        3.210792     3.062590     3.364397     3.770955     2.952177   \n",
       "AACS        3.060751     2.745547     2.157297     3.047658     3.483244   \n",
       "\n",
       "         S008_059_D7  S008_059_D3  S002_050_D7  S002_050_D1  S002_058_D3  \n",
       "gene_id                                                                   \n",
       "A1BG       -0.172061     0.616546    -1.119818    -1.769805    -0.840153  \n",
       "A2M         1.682669     2.031717     2.171863    -0.614794     1.877802  \n",
       "A4GALT     -0.998652    -0.553645     0.792205     2.147521    -1.160755  \n",
       "AAAS        3.503279     3.045054     3.329799     3.158312     3.207962  \n",
       "AACS        2.991366     2.792562     2.559054     2.508396     2.787227  \n",
       "\n",
       "[5 rows x 157 columns]"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.set_index(\"gene_id\", inplace=True)\n",
    "data.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "89e34494-4bfb-4a3b-8db6-23b8536d5152",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>dose</th>\n",
       "      <th>sampling_day</th>\n",
       "      <th>mapped_reads</th>\n",
       "      <th>on_target</th>\n",
       "      <th>arthralgia</th>\n",
       "      <th>pain_in_extremity</th>\n",
       "      <th>pyrexia</th>\n",
       "      <th>myalgia</th>\n",
       "      <th>chills</th>\n",
       "      <th>age</th>\n",
       "      <th>gender</th>\n",
       "      <th>volunteer_id</th>\n",
       "      <th>treatment</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sample_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>S001_045_D1</th>\n",
       "      <td>20000000.0</td>\n",
       "      <td>1</td>\n",
       "      <td>11192441</td>\n",
       "      <td>88.28</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>48</td>\n",
       "      <td>0</td>\n",
       "      <td>001-045</td>\n",
       "      <td>V</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S001_046_D0</th>\n",
       "      <td>20000000.0</td>\n",
       "      <td>0</td>\n",
       "      <td>15527796</td>\n",
       "      <td>85.08</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>39</td>\n",
       "      <td>1</td>\n",
       "      <td>001-046</td>\n",
       "      <td>V</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S001_050_D3</th>\n",
       "      <td>20000000.0</td>\n",
       "      <td>3</td>\n",
       "      <td>10021764</td>\n",
       "      <td>88.52</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>45</td>\n",
       "      <td>0</td>\n",
       "      <td>001-050</td>\n",
       "      <td>V</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S001_053_D0</th>\n",
       "      <td>20000000.0</td>\n",
       "      <td>0</td>\n",
       "      <td>9626092</td>\n",
       "      <td>85.78</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>1</td>\n",
       "      <td>001-053</td>\n",
       "      <td>V</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S001_059_D1</th>\n",
       "      <td>20000000.0</td>\n",
       "      <td>1</td>\n",
       "      <td>10523308</td>\n",
       "      <td>88.57</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>001-059</td>\n",
       "      <td>V</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   dose  sampling_day  mapped_reads  on_target  arthralgia  \\\n",
       "sample_id                                                                    \n",
       "S001_045_D1  20000000.0             1      11192441      88.28           0   \n",
       "S001_046_D0  20000000.0             0      15527796      85.08           0   \n",
       "S001_050_D3  20000000.0             3      10021764      88.52           0   \n",
       "S001_053_D0  20000000.0             0       9626092      85.78           0   \n",
       "S001_059_D1  20000000.0             1      10523308      88.57           0   \n",
       "\n",
       "             pain_in_extremity  pyrexia  myalgia  chills  age  gender  \\\n",
       "sample_id                                                               \n",
       "S001_045_D1                  1        0        0       0   48       0   \n",
       "S001_046_D0                  1        0        0       0   39       1   \n",
       "S001_050_D3                  1        1        0       0   45       0   \n",
       "S001_053_D0                  1        1        1       0   20       1   \n",
       "S001_059_D1                  0        0        0       0   19       0   \n",
       "\n",
       "            volunteer_id treatment  \n",
       "sample_id                           \n",
       "S001_045_D1      001-045         V  \n",
       "S001_046_D0      001-046         V  \n",
       "S001_050_D3      001-050         V  \n",
       "S001_053_D0      001-053         V  \n",
       "S001_059_D1      001-059         V  "
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pheno.set_index(\"sample_id\", inplace=True)\n",
    "pheno.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "8edf627c-3f78-42fb-86ec-613d79762348",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_exprs_data = pd.merge(data.loc[gene_id], pheno[pheno_var], left_index=True, right_index=True).head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "4e0b8bc2-a5ce-4282-9c98-f337fe16c578",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'A1BG': S001_045_D1   -1.533514\n",
      "S001_046_D0   -2.365695\n",
      "S001_050_D3   -1.448913\n",
      "S001_053_D0   -2.187106\n",
      "S001_059_D1   -1.211852\n",
      "Name: A1BG, dtype: float64,\n",
      " 'gender': S001_045_D1    0\n",
      "S001_046_D0    1\n",
      "S001_050_D3    0\n",
      "S001_053_D0    1\n",
      "S001_059_D1    0\n",
      "Name: gender, dtype: int64}\n"
     ]
    }
   ],
   "source": [
    "#pprint(gene_exprs_data.to_dict(\"dict\"))\n",
    "#pprint(gene_exprs_data.to_dict(\"list\"))\n",
    "pprint(gene_exprs_data.to_dict(\"series\"))\n",
    "#pprint(gene_exprs_data.to_dict(\"split\"))\n",
    "#pprint(gene_exprs_data.to_dict(\"records\"))\n",
    "#pprint(gene_exprs_data.to_dict(\"index\"))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "26dc558d-cb39-4649-9fc4-fd0f064181f3",
   "metadata": {},
   "source": [
    "## Gene Correlation Network"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2ef04142-f8ac-4ef5-9cd8-e1522da5285e",
   "metadata": {},
   "source": [
    "**INPUT**: \n",
    "\n",
    "*Gene Coorrelation Matrix*: | source | target | gene correlation |"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "34ba8a91-6222-459f-982b-5e51f9e94660",
   "metadata": {},
   "source": [
    "**OUTPUT** \n",
    "\n",
    "*format: json*\n",
    "\n",
    "nodes : {\n",
    "\n",
    "}\n",
    "\n",
    "edges: {\n",
    "    \n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "806f8312-f547-4e7a-9600-828107c84499",
   "metadata": {},
   "source": [
    "### Volcano Plot\n",
    "**INPUT**: DE table analysis (edgeR)\n",
    "\n",
    "| GeneID | logFC | FDR | Comparison | Group |\n",
    " \n",
    "Surrogate Varibles:\n",
    "- log10P : $-log_{10}(FDR)$\n",
    "- score  : $\\ | -log_{10}(FDR) * logFC\\ |\\ $\n",
    "- DEG    : (-1, 0, +1)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "64f79c07-bd1c-434c-b33d-2e76c4efb2d4",
   "metadata": {},
   "source": [
    "**OUTPUT**\n",
    "*format: json/plot*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "34aec0aa-68e0-4649-983a-fee6ea925e90",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
